packages/x-components/src/store/utils/store-emitters.utils.ts

```diff
@@ -35,17 +35,19 @@ export interface StateSelector<ReturnType, State extends Dictionary, Getters ext
   extends WatchOptions {
   selector: SimpleStateSelector<ReturnType, State, Getters>;
   /**
-   * Checks if the value of the selector has changed.
+   * Asserts if the event should really be emitted taking into account the new and old values and
+   * the module state.
    *
    * @remarks
    * This function exist because Vue will not stop reactivity propagation if the observed variable
    * is an `object`, an `Array`, or the `deep` mode has been enabled.
    *
    * @param newValue - The new value.
    * @param oldValue - The old value.
+   * @param state - The state of the {@link XModule} where this selector is used.
    * @returns True if the value has really changed.
    */
-  filter?(newValue: ReturnType, oldValue: ReturnType): boolean;
+  filter?(newValue: ReturnType, oldValue: ReturnType, state: State): boolean;
```

Comment:
Adding access to the module's state here to be able to filter if the event of the emiter is fired based on the current values of the state.